### PR TITLE
Set pattern for `author` and `version` field in configuration schema.

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

This PR enforces a proper format for the `author` and `version` field in the configuration schemas.

## Fixes

Preliminary work for #155 and #500 

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [X] I have read the [CONTRIBUTING](../master/CONTRIBUTING.rst) doc
- [X] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [X] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/configurations/schemas/aea-config_schema.json [hunk 1]
```diff
@@ -103,47 +103,5 @@
     "description": {
       "type": "string"
     }
-  },
```
Comment (marcofavorito) [inline aea/configurations/schemas/aea-config_schema.json]: we are not going to copy `definitions` for every schema file. It is not maintainable. 
We introduced this workaround in PR #289. However, that is not a long-term solution, so if the problem still occurs on Windows we have to find another solution.

File: aea/configurations/schemas/definitions.json [hunk 2]
```diff
@@ -34,6 +35,54 @@
...
+        }
+      }
+    },
+    "author": {
+      "type": "string",
+      "pattern": "[a-zA-Z0-9_]*"
+    },
+    "version": {
+      "type": "string",
+      "pattern": "^(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$"
```
Comment (marcofavorito) [inline aea/configurations/schemas/definitions.json]: We use semantic versioning for version strings (not backward compatible):
[https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string](https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string)

File: aea/configurations/schemas/definitions.json [hunk 3]
```diff
@@ -34,6 +35,54 @@
...
+    },
+    "public_id": {
+      "type": "string",
+      "pattern": "[a-zA-Z_][a-zA-Z0-9_]*"
```
Comment (marcofavorito) [inline aea/configurations/schemas/definitions.json]: This is not used atm, to be removed.

File: aea/configurations/schemas/definitions.json [hunk 4]
```diff
@@ -34,6 +35,54 @@
...
+          "type": "string"
+        },
+        "port": {
+          "type": "number"
+        }
+      }
+    },
+    "author": {
+      "type": "string",
+      "pattern": "[a-zA-Z0-9_]*"
```
Comment (marcofavorito) [inline aea/configurations/schemas/definitions.json]: Author `""` is allowed in the configuration file for backward compatibility. However, it is not going to work when interacting with the registry. 

That suggests `aea scaffold` sets the `author` field according to the session kept by the CLI tool (if any, eventually).

File: aea/cli/create.py [hunk 5]
```diff
@@ -75,7 +75,7 @@ def create(click_context, agent_name):
         # create a config file inside it
         logger.info("Creating config file {}".format(DEFAULT_AEA_CONFIG_FILE))
         config_file = open(os.path.join(agent_name, DEFAULT_AEA_CONFIG_FILE), "w")
-        agent_config = AgentConfig(agent_name=agent_name, aea_version=aea.__version__, author="", version="v1", license="", fingerprint="", url="", registry_path=DEFAULT_REGISTRY_PATH, description="")
+        agent_config = AgentConfig(agent_name=agent_name, aea_version=aea.__version__, author="", version="1.0.0", license="", fingerprint="", url="", registry_path=DEFAULT_REGISTRY_PATH, description="")
```
Comment (DavidMinarsch) [inline aea/cli/create.py]: I think the default version should be `0.1.0`. We should not suggest that a default agent project is somehow a v1.

File: aea/configurations/schemas/definitions.json (hunk 4 above)
Comment (DavidMinarsch) [reply aea/configurations/schemas/definitions.json]: Ok

File: aea/configurations/schemas/definitions.json (hunk 2 above)
Comment (DavidMinarsch) [reply aea/configurations/schemas/definitions.json]: Sounds good

File: aea/configurations/schemas/definitions.json (hunk 3 above)
Comment (DavidMinarsch) [reply aea/configurations/schemas/definitions.json]: Ok so remove..?

File: aea/cli/create.py (hunk 5 above)
Comment (marcofavorito) [reply aea/cli/create.py]: done

**Pull request merged**